internals-js/src/supergraphs.ts

```diff
@@ -118,7 +123,7 @@ export class Supergraph {
       // Note that `extractSubgraphsFromSupergraph` redo a little bit of work we're already one, like validating
       // the supergraph. We could refactor things to avoid it, but it's completely negligible in practice so we
       // can leave that to "some day, maybe".
-      this._subgraphs = extractSubgraphsFromSupergraph(this.schema);
+      this._subgraphs = extractSubgraphsFromSupergraph(this.schema, !this.assumeValid);
```

Comment:
I don't love that we start with `validateReceivedSupergraphs` then go to `assumeValid` then back to `validateExtractedSubgraphs`. It would be nice to not keep negating the poor boolean.